src/Set.ts

```diff
@@ -31,6 +31,21 @@ export const some = <A>(x: Set<A>, predicate: Predicate<A>): boolean => {
   return found
 }
 
+/**
+  * Projects a Set through a function
+  *  @function */
+export const map = <A, B>(x: Set<A>, bset: Setoid<B>, f: (x: A) => B): Set<B> => {
+  const ismember = member(bset)
+  const r = new Set<B>()
+  x.forEach(e => {
+    const v = f(e)
+    if (!ismember(r)(v)) {
```

Comment:
Maybe we can do less work here defining `ismember` as

```ts
const r = new Set<B>()
const ismember = member(bset)(r)
```

Reply:
Indeed, also I'm not sure what to favor between using `forEach`or using an Iterator, do you have any clue?